scio-core/src/main/scala/com/spotify/scio/coders/Coder.scala

```diff
@@ -224,12 +202,42 @@ private[scio] case class WrappedBCoder[T](u: BCoder[T]) extends BCoder[T] {
     u.registerByteSizeObserver(value, observer)
 }
 
-private object WrappedBCoder {
+private[scio] object WrappedBCoder {
   def create[T](u: BCoder[T]): BCoder[T] =
     u match {
       case WrappedBCoder(_) => u
       case _                => new WrappedBCoder(u)
     }
+
+  def appendMaterializationStack[X <: Throwable](
```

Comment:
I think I would keep this logic in a separate `object` ... I kinda like to keep the decoupling from `WrappedBCoder`

Reply:
Good point, will pull that away in a StackTraceUtil of sorts